### PR TITLE
Dynamodb

### LINKED ISSUE #1811: Add DynamoDB connectivity
Apache MetaModel already has the module - we just need to plug it in.

### PR DESCRIPTION
**Pull request by kaspersorensen**
This makes DataCleaner capable of querying and writing to AWS DynamoDB.

Fixes #1811 

A screenshot for reference:

image

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/windows/DynamoDbDatastoreDialog.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    private final JXTextField _accessKeyField;
+    private final JPasswordField _accessSecretField;
+    private final TableDefinitionOptionSelectionPanel _tableDefinitionWidget;
+
+    @Inject
+    public DynamoDbDatastoreDialog(final WindowContext windowContext, final MutableDatastoreCatalog catalog,
+            @Nullable final DynamoDbDatastore originalDatastore, final UserPreferences userPreferences) {
+        super(originalDatastore, catalog, windowContext, userPreferences);
+
+        _regionField = new DCComboBox<>(Regions.values());
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/DynamoDbDatastoreDialog.java]: Didn't we nuke that silly _ with checkstyle, or do I remember wrong?

Comment (kaspersorensen) [reply desktop/ui/src/main/java/org/datacleaner/windows/DynamoDbDatastoreDialog.java]: I am not aware that we nuked it, and I largely created this Dialog class by copy-pasting another dialog and fitting it to this type of datastore. The one I copied from had the underscores so I gathered that it was still a used convention :-)

File: desktop/ui/src/main/java/org/datacleaner/windows/DynamoDbDatastoreDialog.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+import com.amazonaws.regions.Regions;
+
+public class DynamoDbDatastoreDialog extends AbstractDatastoreDialog<DynamoDbDatastore> implements SchemaFactory {
+
+    private static final long serialVersionUID = 1L;
+
+    private final DCComboBox<Regions> _regionField;
+    private final JXTextField _accessKeyField;
+    private final JPasswordField _accessSecretField;
```
[the remark this answers is not in the document]
Comment (kaspersorensen) [reply desktop/ui/src/main/java/org/datacleaner/windows/DynamoDbDatastoreDialog.java]: You're right, the names in AWS's console is:

* Access key ID
* Secret access key

I'll change it accordingly

File: desktop/ui/src/main/java/org/datacleaner/windows/DynamoDbDatastoreDialog.java (hunk 1 above)
Comment (LosD) [reply desktop/ui/src/main/java/org/datacleaner/windows/DynamoDbDatastoreDialog.java]: I probably remember wrongly, then :-)

**Pull request merged**